packages/@react-aria/collections/src/CollectionBuilder.tsx

```diff
@@ -127,22 +127,40 @@ function useCollectionDocument<T extends object, C extends BaseCollection<T>>(cr
 
 const SSRContext = createContext<BaseNode<any> | null>(null);
 
-function useSSRCollectionNode<T extends Element>(Type: string, props: object, ref: ForwardedRef<T>, rendered?: any, children?: ReactNode, render?: (node: Node<T>) => ReactElement) {
+export type CollectionNodeClass<T> = {
+  new (key: Key): CollectionNode<T>,
+  readonly type: string
+};
+
+function createCollectionNodeClass(type: string): CollectionNodeClass<any> {
+  let NodeClass = function (key: Key) {
+    return new CollectionNode(type, key);
+  } as any;
+  NodeClass.type = type;
+  return NodeClass;
```

Comment:
if we address the above, maybe we can remove the need for this
or simplify it, i'm not sure why it's wrapped in a function